app/models/user.rb

```diff
@@ -1,6 +1,10 @@
-# class User < ActiveRecord::Base
-#   has_secure_password
-# end
+class User < ActiveRecord::Base
+  has_secure_password
+
+  validates :email, presence: true, uniqueness: true, format: { with: /\A[^@\s]+@([^@\s]+\.)+[^@\s]+\z/, message: "is not a valid email address" }
```

Comment:
It would be better to use `URI::MailTo::EMAIL_REGEXP` than to write your own RegEx here.